### PR TITLE
chore: Typescript Cleanup Part 2 of 3 for TabId error

### PR DESCRIPTION
**Pull request by v-prachin**
#### Details

Typescript Cleanup Part 2 of 3 
We have total 140 errors in 104 files. Thus, targeting each error and fixing it. 

##### Motivation
Existing User Story: [User Story: 2132090](https://dev.azure.com/mseng/1ES/_workitems/edit/2150742)

Tab Id error in below files observed after removing deprecated properties from tsconfig file:

1.src/background/extension-details-view-controller.ts
2.src/tests/unit/tests/background/extension-details-view-controller.test.ts
3.src/background/feature-flags-controller.ts
4.src/background/user-configuration-controller.ts
5.src/tests/unit/tests/background/feature-flags-controller.test.ts
6.src/tests/unit/tests/background/user-configuration-controller.test.ts
7.src/background/keyboard-shortcut-handler.ts
8.src/background/target-page-controller.ts
9.src/tests/unit/tests/background/background-message-distributor.test.ts
10.src/background/keyboard-shortcut-handler.test,ts
11.src/background/target-page-controller.test.ts

We already have a Message interface which has two components messageType & payload. InterpreterMessage is a type which has Message & TabId. As TabId was not used so often in Message thus this change must have be done by someone (I assume). So as part of TabId error, replacing Message with InterpreterMessage.

#### Pull request checklist

- [x] Addresses an existing issue: [User Story: 2132090](https://dev.azure.com/mseng/1ES/_workitems/edit/2150742)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/common/message.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export interface Message {
     payload?: any;
 }
 
-export type InterpreterMessage = Message & { tabId?: number };
+export type InterpreterMessage = Message & { tabId?: any };
```
Comment (v-viyada) [inline src/common/message.ts:9]: Keep the type to number and made related changes in other files

**Review by madalynrose — CHANGES_REQUESTED** [review]
I'm fairly certain we want to find a solution that preserves `tabId` in as many places as possible. I would recommend *not* removing `tabId` from any objects and instead turning any `Message` objects that are causing errors into `InterpreterMessage`s.

File: src/background/user-configuration-controller.ts [hunk 2]
```diff
@@ -25,10 +24,9 @@ export class UserConfigurationController {
         const payload: SetTelemetryStatePayload = {
             enableTelemetry,
         };
-        const message: Message = {
+        const message: InterpreterMessage = {
             messageType: Messages.UserConfig.SetTelemetryConfig,
             payload: payload,
-            tabId: null,
```
Comment (v-rakeshsh) [inline src/background/user-configuration-controller.ts:31]: guess we need tabId here?

Comment (v-prachin) [reply src/background/user-configuration-controller.ts:31]: We removed tabId from here as was getting yarn null check error
NullCheck

File: src/common/message.ts (hunk 1 above)
Comment (v-prachin) [reply src/common/message.ts]: Made the suggested change

**Pull request merged**